Sources/App/Core/Query+Support/Joined3+Package.swift

```diff
@@ -40,4 +40,12 @@ extension Joined3 where M == Package, R1 == Repository, R2 == Version {
             .filter(Repository.self, \.$owner, .custom("ilike"), owner)
             .filter(Repository.self, \.$name, .custom("ilike"), repository)
     }
+    
+    static func query(on database: Database, packageId: Package.Id) -> JoinedQueryBuilder<Self> {
```

Comment:
I'll take @finestructure's view on this as he's the keeper of the `Joined` classes, but I wonder if this might need a more descriptive name. It makes sense to query with the packageID, but the fact that this is filtering to the default branch version too is a little hidden from whoever is calling this function.

Alternatively, we could move the filter to outside this query and let the calling code add that filter.

Don't make any changes based on what I say to this, though as I know @finestructure will have a view on it and we should go with his recommendation.